chiselFrontend/src/main/scala/chisel3/core/Bits.scala

```diff
@@ -106,13 +130,8 @@ sealed abstract class Bits(width: Width, override val litArg: Option[LitArg])
     if (x < 0) {
       Builder.error(s"Negative bit indices are illegal (got $x)")
     }
-    if (isLit()) {
-      (((litValue() >> x.toInt) & 1) == 1).asBool
-    } else {
-
-      requireIsHardware(this, "bits to be indexed")
-      pushOp(DefPrim(sourceInfo, Bool(), BitsExtractOp, this.ref, ILit(x), ILit(x)))
-    }
```

Comment:
Just for the historical record:

These are minor constant propagations done on literals. I did some archaeology to understand why they were there in the first place. They date to very early chisel3 https://github.com/freechipsproject/chisel3/commit/7ada20e324ca75116441d00b575ed217c41cc1c8 and appear to be largely to work around bugs in order to get riscv-mini compiling. Most of them have already been removed but a couple seem to have stuck around. I agree they should be removed.